internal/uvm/create.go

```diff
@@ -428,10 +435,22 @@ func Create(opts *UVMOptions) (_ *UtilityVM, err error) {
 		}
 
 		kernelArgs += ` pci=off -- ` + initArgs
-		vm.Chipset.Uefi.BootThis = &hcsschema.UefiBootEntry{
-			DevicePath:   `\` + opts.KernelFile,
-			DeviceType:   "VmbFs",
-			OptionalData: kernelArgs,
+
+		if !opts.KernelDirect {
+			vm.Chipset.Uefi.BootThis = &hcsschema.UefiBootEntry{
+				DevicePath:   `\` + opts.KernelFile,
+				DeviceType:   "VmbFs",
+				OptionalData: kernelArgs,
+			}
+		} else {
+			vm.Chipset.Uefi = nil
+			vm.Chipset.LinuxKernelDirect = &hcsschema.LinuxKernelDirect{
+				KernelFilePath: filepath.Join(opts.BootFilesPath, opts.KernelFile),
+				KernelCmdLine:  kernelArgs,
```

Comment:
In this case we should not pass the initrd= path above.

Reply:
We wouldn't? I did refactor this to latest so its possible I messed up but:

```go
kernelArgs := "initrd=/" + opts.RootFSFile
if actualRootFSType == PreferredRootFSTypeVHD {
    kernelArgs = "root=/dev/pmem0 init=/init"
}
```
If is `PreferredRootFSTypeVHD` `kernelArgs` will not contain `initrd=`
```go
vm.Chipset.LinuxKernelDirect = &hcsschema.LinuxKernelDirect{
        KernelFilePath: filepath.Join(opts.BootFilesPath, opts.KernelFile),
        KernelCmdLine:  kernelArgs,
    }
    if actualRootFSType == PreferredRootFSTypeInitRd {
        vm.Chipset.LinuxKernelDirect.InitRdPath = filepath.Join(opts.BootFilesPath, opts.RootFSFile)
    }
}
```
and here if it is `PreferredRootFSTypeInitRd` `kernelArgs` will contain `initrd=` so we tell the product where to find the file.

Reply:
Let me get this on a build to test again though

Reply:
I believe the initrd= part is only used by the EFI loader in the Linux kernel, so when we skip it, it gets the initrd info from the Linux direct loader, not the kernel command line.

Reply:
You are correct no need for `initrd=` in KernelDirect